emcmake.py

```diff
@@ -12,25 +12,47 @@
 from tools import utils
 from subprocess import CalledProcessError
 
+# Print message to stderr and exit early.
+def error(message, exit_code=1, file=sys.stderr):
+  print(message, file=file)
+  sys.exit(exit_code)
 
-#
-# Main run() function
-#
-def run():
-  if len(sys.argv) < 2 or sys.argv[1] in ('--version', '--help'):
-    print('''\
+# Conditionally add a toolchain and run emulator.
+def create_command():
+  if len(sys.argv) < 2 or sys.argv[1] in ('--help', '--version'):
+    error('''\
 emcmake is a helper for cmake, setting various environment
 variables so that emcc etc. are used. Typical usage:
 
   emcmake cmake [FLAGS]
-''', file=sys.stderr)
-    return 1
+''')
+
+  unsupported = {
+    '--build',
+    '--install',
+    '--open',
+    '-E',
+    '--find-package',
+    '--help',
+    '--version',
+  }
 
   args = sys.argv[1:]
+  if len(args) < 2 or args[1] in unsupported:
+    return args
 
   def has_substr(args, substr):
     return any(substr in s for s in args)
 
+  # Check if it's called as a script argument, which is an unsupported case.
+  # Script arguments can lead with as many `-D` cases, and then
+  # have a `-P`.
+  index = 1
+  while index < len(args) and args[index].startswith('-D'):
+    index += 1
+  if index < len(args) and args[index].startswith('-P'):
+    return args
```

Comment:
How about just:

```
if '-P' in args:
   return args
```

In fact, why not just add `-P` to unsupported above and avoid this block?

Reply:
Sure, we can do that. I'll update the PR now.